graphdatascience/tests/integrationV2/procedure_surface/arrow/catalog/test_node_properties_arrow_endpoints.py

```diff
@@ -0,0 +1,125 @@
+from typing import Generator
+
+import pytest
+
+from graphdatascience import Graph, QueryRunner
+from graphdatascience.arrow_client.authenticated_flight_client import AuthenticatedArrowClient
+from graphdatascience.arrow_client.v2.remote_write_back_client import RemoteWriteBackClient
+from graphdatascience.procedure_surface.api.catalog.node_properties_endpoints import NodePropertySpec
+from graphdatascience.procedure_surface.arrow.catalog.node_properties_arrow_endpoints import (
+    NodePropertiesArrowEndpoints,
+)
+from graphdatascience.tests.integrationV2.procedure_surface.arrow.graph_creation_helper import (
+    create_graph,
+    create_graph_from_db,
+)
+
+
+@pytest.fixture
+def sample_graph(arrow_client: AuthenticatedArrowClient) -> Generator[Graph, None, None]:
+    gdl = """
+        CREATE
+        (a: Node {prop1: 1, prop2: 42.0}),
+        (b: Node {prop1: 2, prop2: 43.0}),
+        (c: Node {prop1: 3, prop2: 44.0})
+    """
+
+    with create_graph(arrow_client, "g", gdl) as G:
+        yield G
+
+
+@pytest.fixture
+def db_graph(arrow_client: AuthenticatedArrowClient, query_runner: QueryRunner) -> Generator[Graph, None, None]:
+    graph_data = """
+        CREATE
+        (a: Node {prop1: 1, prop2: 42.0}),
+        (b: Node {prop1: 2, prop2: 43.0}),
+        (c: Node {prop1: 3, prop2: 44.0})
+    """
+
+    with create_graph_from_db(
+        arrow_client,
+        query_runner,
+        "g",
+        graph_data,
+        "MATCH (n) WITH gds.graph.project.remote(n, null, {sourceNodeLabels: labels(n), targetNodeLabels: null, sourceNodeProperties: properties(n)}) as g RETURN g",
+    ) as G:
+        yield G
+
+
+@pytest.fixture
+def node_properties_endpoints(
+    arrow_client: AuthenticatedArrowClient,
+) -> Generator[NodePropertiesArrowEndpoints, None, None]:
+    yield NodePropertiesArrowEndpoints(arrow_client)
+
+
+@pytest.fixture
+def node_properties_endpoints_with_db(
+    arrow_client: AuthenticatedArrowClient, query_runner: QueryRunner
+) -> Generator[NodePropertiesArrowEndpoints, None, None]:
+    yield NodePropertiesArrowEndpoints(arrow_client, RemoteWriteBackClient(arrow_client, query_runner))
+
+
+def test_stream_node_properties(node_properties_endpoints: NodePropertiesArrowEndpoints, sample_graph: Graph) -> None:
+    result = node_properties_endpoints.stream(G=sample_graph, node_properties=["prop1", "prop2"])
+
+    assert len(result) == 3
+    assert "nodeId" in result.columns
+    assert "prop1" in result.columns
+    assert "prop2" in result.columns
```

Comment:
nit: i prefer to check for 
```suggestion
    assert len(result) == 3
    assert set(result.columns) == {"nodeId", "prop1", "prop2"}
```